operator/internal/status/status.go

```diff
@@ -45,7 +45,8 @@ func Refresh(ctx context.Context, k k8s.Client, req ctrl.Request, now time.Time,
 	statusUpdater := func(stack *lokiv1.LokiStack) {
 		stack.Status.Components = *cs
 		stack.Status.Conditions = mergeConditions(stack.Status.Conditions, activeConditions, metaTime)
-		stack.Status.Storage.CredentialMode = credentialMode
+		stack.Status.Storage.CredentialMode = statusInfo.Storage
+		stack.Status.NetworkPolicies = statusInfo.NetworkPolicies
```

Comment:
This was a subtle bug before (because of the empty value of `credentialMode`), but when LokiStack is degraded then `statusInfo` will now be `nil` causing this code to panic.